test/dynamics/pulse/test_pulse_to_signals.py

```diff
@@ -303,15 +317,101 @@ def test_multiple_channels_with_gaps(self):
 
-        self.assertAllClose(signals[0].samples, samples0, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[1].samples, samples1, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[2].samples, samples2, atol=1e-14, rtol=1e-14)
-        self.assertAllClose(signals[3].samples, samples3, atol=1e-14, rtol=1e-14)
+        self.assertAllClose(signals[0].samples, samples0, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[1].samples, samples1, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[2].samples, samples2, atol=1e-7, rtol=1e-7)
+        self.assertAllClose(signals[3].samples, samples3, atol=1e-7, rtol=1e-7)
         self.assertTrue(signals[0].carrier_freq == 5.0)
         self.assertTrue(signals[1].carrier_freq == 3.1)
         self.assertTrue(signals[2].carrier_freq == 0.0)
         self.assertTrue(signals[3].carrier_freq == 4.0)
 
+    def test_InstructionToSignals(self):
+        """Test InstructionToSignals with get samples function."""
+        gauss_get_waveform_samples = (
+            pulse.Gaussian(duration=5, amp=0.983, sigma=2.0).get_waveform().samples
+        )
+        with pulse.build() as schedule:
+            pulse.play(pulse.Gaussian(duration=5, amp=0.983, sigma=2.0), pulse.DriveChannel(0))
+        converter = InstructionToSignals(dt=self._dt, channels=["d0"])
+        signals = converter.get_signals(block_to_schedule(schedule))
+        self.assertAllClose(signals[0].samples, gauss_get_waveform_samples, atol=1e-7, rtol=1e-7)
+
+
+class TestJaxGetSamples(QiskitDynamicsTestCase, TestJaxBase):
+    """Tests get_samples function by using Jax."""
+
+    def setUp(self):
+        """Set up gaussian waveform samples for comparison."""
+        self.gauss_get_waveform_samples = (
+            pulse.Gaussian(duration=5, amp=0.983, sigma=2.0).get_waveform().samples
+        )
+        self.constant_get_waveform_samples = (
+            pulse.Constant(duration=5, amp=0.1).get_waveform().samples
+        )
+        self._dt = 0.222
+
+    def test_jit_get_samples(self):
+        """Test compiling to get samples of Pulse."""
+
+        def jit_func_get_samples(amp):
+            parameters = {"amp": amp}
+            _time, _amp, _duration = sym.symbols("t, amp, duration")
+            envelope_expr = _amp * sym.Piecewise(
+                (1, sym.And(_time >= 0, _time <= _duration)), (0, True)
+            )
+            valid_amp_conditions_expr = sym.Abs(_amp) <= 1.0
+            # we can use only SymbolicPulse when jax-jitting
+            # bacause jax-jitting doesn't correspond to validate_parameters in qiskit.pulse.
+            instance = SymbolicPulse(
+                pulse_type="Constant",
+                duration=5,
+                parameters=parameters,
+                envelope=envelope_expr,
+                valid_amp_conditions=valid_amp_conditions_expr,
+            )
+            return get_samples(instance)
```

Comment:
E.g. this test should use `InstructionToSignals` instead of `get_samples` directly.